plugins/modules/clustering/znode.py

```diff
@@ -49,6 +49,18 @@
             - Recursively delete node and all its children.
         type: bool
         default: false
+    scheme:
+        description:
+            - 'Authentication scheme.'
+        choices: [ digest, sasl ]
+        type: str
+        default: "digest"
+        required: false
+    credential:
+        description:
+            - 'The credential value. Depends on scheme (format "digest": user:password, "sasl": user:password).'
+        type: str
```

Comment:
I'm wondering whether it's better to provide a dict here, respectively have two options `username` and `password` instead of one (`credential`).

Reply:
To be honest i don't know nothing about sasl but reading the wiki about sasl i can imagine that it does not depend on user and password but some other kind of secret based on the actual used sasl implmentation. I'm just exposing the underlying kazoo api here which also doen't makt this distinction. Besides that: this would leave us with a user password dict in case of digest and the credentail option in case of sasl. This might confuse people and increases the code to handle each case. I assume that users of zookeeper are familiar with this kind of notation.

Reply:
:shrug: I'm not using zookeeper, so no idea. If nobody else chimes in who knows anything about zookeeper, let's keep this as you suggested :)